### PR TITLE
Increase default VNC timeout for remote connections

### PR DESCRIPTION
**Pull request by Martchus**
* Make timeout configurable as well
* See https://progress.opensuse.org/issues/76813

### CODE REVIEW

File: t/27-consoles-vnc.t [hunk 1]
```diff
@@ -50,4 +52,23 @@ subtest 'handling VNC stall, malformed RFB protocol on re-connect' => sub {
...
+    _setup_rfb_magic;
+    combined_like {
+        throws_ok { $c->login } 'OpenQA::Exception::VNCSetupError', 'dies on connect timeout' }
+    qr/.*Error connecting to.*/, 'error logged';
+    is $attempts, 7, 'login attempts for local hostname';
+    $attempts = 0;
+    $c->hostname('10.161.145.95');
+    combined_like {
+        throws_ok { $c->login } 'OpenQA::Exception::VNCSetupError', 'dies on connect timeout (2)'
+    } qr/.*Error connecting to.*/, 'error logged (2)';
```
Comment (okurz) [inline t/27-consoles-vnc.t:70]: Is this trying to actually connect to this adress? I would prefer to mock the real connection attempt or at least use a non-reachable localhost address. What about 127.0.0.2?

File: consoles/VNC.pm [hunk 2]
```diff
@@ -135,7 +134,9 @@ sub login ($self, $connect_timeout = undef) {
     my $hostname = $self->hostname || 'localhost';
     my $port = $self->port || 5900;
     my $description = $self->description || 'VNC server';
-
+    my $local_timeout = $bmwqemu::vars{VNC_CONNECT_TIMEOUT_LOCAL} // 10;
+    my $remote_timeout = $bmwqemu::vars{VNC_CONNECT_TIMEOUT_REMOTE} // 60;
+    $connect_timeout //= ($hostname =~ qr/(localhost|127\.0\.0\.\d+|::1)/) ? $local_timeout : $remote_timeout;
```
Comment (Martchus) [inline consoles/VNC.pm:139]: @kraih Does Perl or some module like Mojolicious provide a helper for this distinction? (e.g. something in the lines of https://github.com/Martchus/syncthingtray/blob/8e6bd62030db1867fd68d3798a7c1e497933bae5/connector/utils.cpp#L118)

File: t/27-consoles-vnc.t (hunk 1 above)
Comment (okurz) [reply t/27-consoles-vnc.t:70]: ok, cool

**Review by kalikiana — APPROVED** [review]
Looks sensible enough to me!

File: consoles/VNC.pm (hunk 2 above)
Comment (kraih) [reply consoles/VNC.pm:139]: This should probably be a utility function with tests.

**Review by okurz — APPROVED** [review]

**Pull request merged**